src/Step/Deploy/InstallUpdate/ConfigUpdate/DbConnection.php

```diff
@@ -60,82 +61,272 @@ class DbConnection implements StepInterface
-     * @param RelationshipConnectionFactory $connectionFactory
+     * @param RelationshipConnectionFactory $connectionDataFactory
      * @param LoggerInterface $logger
+     * @param FlagManager $flagManager
      */
     public function __construct(
         DeployInterface $stageConfig,
-        MergedConfig $mergedConfig,
+        DbConfig $dbConfig,
         ResourceConfig $resourceConfig,
         ConfigWriter $configWriter,
         ConfigReader $configReader,
         ConfigMerger $configMerger,
-        RelationshipConnectionFactory $connectionFactory,
-        LoggerInterface $logger
+        RelationshipConnectionFactory $connectionDataFactory,
+        LoggerInterface $logger,
+        FlagManager $flagManager
     ) {
         $this->stageConfig = $stageConfig;
-        $this->mergedConfig = $mergedConfig;
+        $this->dbConfig = $dbConfig;
         $this->resourceConfig = $resourceConfig;
         $this->configWriter = $configWriter;
         $this->configReader = $configReader;
         $this->configMerger = $configMerger;
         $this->logger = $logger;
-        $this->connectionFactory = $connectionFactory;
+        $this->connectionDataFactory = $connectionDataFactory;
+        $this->flagManager = $flagManager;
     }
 
     /**
      * @inheritdoc
      */
     public function execute()
     {
-        $config = $this->configReader->read();
+        $dbConfig = $this->dbConfig->get();
+
+        if (!isset($dbConfig[DbConfig::KEY_CONNECTION])) {
+            /**
+             * Is calling only in case when database relationship configuration doesn't exist and
+             * database is not configured through .magento.env.yaml or env variable.
+             * It's workaround for scenarios when magento was installed by raw setup:install command
+             * not by deploy scripts.
+             */
+            $this->logger->notice(
+                'Database relationship configuration doesn\'t exist'
+                . ' and database is not configured through .magento.env.yaml or env variable.'
+                . ' Will be applied the previous database configuration.'
+            );
+            return;
+        }
 
         $this->logger->info('Updating env.php DB connection configuration.');
-        $config['db'] = $this->mergedConfig->get();
-        $config['resource'] = $this->resourceConfig->get();
 
-        $this->addLoggingAboutSlaveConnection();
-        $this->configWriter->create($config);
+        $mageConfig = $this->configReader->read();
+        $mageSplitConnectionsConfig = array_intersect_key(
+            $mageConfig[DbConfig::KEY_DB][DbConfig::KEY_CONNECTION],
+            array_flip(DbConfig::SPLIT_CONNECTIONS)
+        );
+
+        $isCustomDefaultConnection = !$this->isSameConnection(
+            $dbConfig[DbConfig::KEY_CONNECTION][DbConfig::CONNECTION_DEFAULT],
+            $mageConfig[DbConfig::KEY_DB][DbConfig::KEY_CONNECTION][DbConfig::CONNECTION_DEFAULT]
```

Comment:
seems like there may be case when connection name may have another name than 'default', e.g. there is a custom connection name in **resource** configuration . Need to create separate task to investigate it.

Reply:
Checked.
If the connection with the name "default" is missing Magento does not work